sql: add assertion for txn passed to internal executor

ZhouXing19 opened the pull request:
If an internal executor is created with a txn binding to it, it should not be used to execute a statement with a nil txn.

Release note: None

ZhouXing19 committed:
sql/internalExecutor: add assertion for txn

If a internal executor is created with a txn binding to it, it should not be
used to execute statement with a nil txn.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88295)
<!-- Reviewable:end -->

ZhouXing19 committed:
upgrades: fix usage of internal executor in findNextTableToUpgrade

Previously, the internal executor always run queries with a nil txn, even though
it's constructed with a txn binding to it.

We now change it to pass the txn to the internal executor in
`findNextTableToUpgrade()`

Release note: None

ZhouXing19 commented:
RFAL, thanks!

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88295#-:-NCvTSU6BAUZ4WHa4aWQ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ZhouXing19)



<!-- Sent from Reviewable.io -->

ZhouXing19 commented:
Thanks!
bors r=ajwerner
